make/source.mk

```diff
@@ -219,8 +219,10 @@ SPEED_OPTIMISED_SRC := $(SPEED_OPTIMISED_SRC) \
             drivers/buf_writer.c \
             drivers/bus.c \
             drivers/bus_spi.c \
+            drivers/bus_spi_ll.c \
             drivers/exti.c \
             drivers/io.c \
+            drivers/max7456.c \
```

Comment:
Does this _have_ to be speed optimised? (Thinking of the scarceness of flash space on F3.)

Reply:
Good point. Glad you noticed it.

This particular driver
- has a large fixed for loop (100) which is a nice target for loop unrolling, but the side effect would be horrible if it is actually unrolled.
- called at 60Hz which is not too often, and
- speed is dominated by SPI transaction speed (when not using DMA).

So, I would say it shouldn't be optimized for F3, but not sure about F4 and F7.

The point I wanted to raise with this particular entry is whether to introduce per MCU optimization list or not.

Reply:
Good point, we could definitely move to having different optimisation sets again for F3 / F4 / F7, like we had originally. I think we made everything 'F3 or better' use the F3 set since it was assumed that not optimising non-time-critical parts would not give us an advantage on F4 (or now F7).